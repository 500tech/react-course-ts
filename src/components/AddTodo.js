import React, { Component, createRef } from 'react';

export class AddTodo extends Component {
  state = {
    draft: '',
  };

  componentDidMount() {
    this._input.current.focus();
  }

  _input = createRef();

  onDraftChange = event => {
    const { target } = event;
    const { value } = target;
    this.setState({ draft: value });
  };

  addTodo = () => {
    this.props.onAddTodo(this.state.draft);
    this.setState({ draft: '' });
  };

  render() {
    return (
      <div>
        <input
          ref={this._input}
          value={this.state.draft}
          onChange={this.onDraftChange}
        />
        <button disabled={this.state.draft === ''} onClick={this.addTodo}>
          Add Me!
        </button>
      </div>
    );
  }
}
