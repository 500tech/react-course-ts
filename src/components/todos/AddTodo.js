import React, { Component, createRef } from 'react';
import { ReduxConsumer } from '../../ReduxBridge';
import { increment } from '../../state/actions/count.actions';

export class AddTodo extends Component {
  state = { text: '' };

  input = createRef();

  componentDidMount() {
    // this.input.current.focus();
  }

  setText = text => this.setState({ text });

  reset = () => this.setText('');

  onTextChange = event => this.setText(event.target.value);

  onAddTodo = () => {
    this.props.addTodo(this.state.text);
    this.reset();
  };

  render() {
    return (
      <ReduxConsumer>
        {({ dispatch }) => (
          <>
            <input
              ref={this.input}
              onChange={this.onTextChange}
              value={this.state.text}
            />
            <button
              disabled={!this.state.text.length}
              onClick={() => {
                dispatch(increment());
                this.onAddTodo();
              }}
            >
              Add
            </button>
          </>
        )}
      </ReduxConsumer>
    );
  }
}
