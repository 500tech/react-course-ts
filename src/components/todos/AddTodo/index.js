import { connect } from 'react-redux';
import { AddTodo as BaseAddTodo } from './AddTodo';
import { increment } from '../../../state/actions/count.actions';

const mapStateToProps = () => ({});
function mapDispatchToProps(dispatch) {
  return {
    increment() {
      dispatch(increment());
    },
  };
}

export const AddTodo = connect(
  mapStateToProps,
  mapDispatchToProps
)(BaseAddTodo);
