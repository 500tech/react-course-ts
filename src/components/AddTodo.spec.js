import React from 'react';
import { shallow, mount } from 'enzyme';
import AddTodo from './AddTodo';
import toJson from 'enzyme-to-json';

const isFocusedElement = wrapper =>
  wrapper.getDOMNode() === document.activeElement;

describe('<AddTodo />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<AddTodo />);
    expect(wrapper.find('input').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
  });
  it('autofocuses if needed', () => {
    const wrapper = mount(<AddTodo />);
    expect(isFocusedElement(wrapper.find('input'))).toBe(false);
    wrapper.unmount();
    wrapper.setProps({ autoFocus: true });
    wrapper.mount();
    expect(isFocusedElement(wrapper.find('input'))).toBe(true);
    wrapper.unmount();
  });
  it('has a text value reflecting of its state', () => {
    const wrapper = shallow(<AddTodo />);
    wrapper
      .find('input')
      .simulate('change', { target: { value: 'Do a thing' } });
    expect(wrapper.find('input').props().value).toBe('Do a thing');
  });
  it('is sane', () => {
    const wrapper = shallow(<AddTodo />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
