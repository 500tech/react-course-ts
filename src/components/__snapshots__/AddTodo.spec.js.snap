// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddTodo /> is sane 1`] = `
<Fragment>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="What should I do?"
    type="text"
    value=""
  />
  <button
    disabled={true}
    onClick={[Function]}
  >
    Add
  </button>
</Fragment>
`;
